do_div needs 64 bit type on all arches

upstream report: https://www.spinics.net/lists/netfilter-devel/msg70467.html

diff --git a/extensions/pknock/xt_pknock.c b/extensions/pknock/xt_pknock.c
index 5296eef..3c304e0 100644
--- a/extensions/pknock/xt_pknock.c
+++ b/extensions/pknock/xt_pknock.c
@@ -338,7 +338,7 @@ is_interknock_time_exceeded(const struct peer *peer, unsigned int max_time)
 static inline bool
 has_logged_during_this_minute(const struct peer *peer)
 {
-	unsigned long x, y;
+	uint64_t x, y;
 	if (peer == NULL)
 		return 0;
 	x = ktime_get_seconds();
@@ -717,7 +717,7 @@ has_secret(const unsigned char *secret, unsigned int secret_len, uint32_t ipsrc,
 	unsigned int hexa_size;
 	int ret;
 	bool fret = false;
-	unsigned long x;
+	uint64_t x;
 	unsigned int epoch_min;
 
 	if (payload_len == 0)
